.github/workflows/scripts/install-go.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+set -x
+
+GO_VERSION=1.22.0
+ARCH=`uname -i`
+
+if [ $ARCH -eq "x86_64" ]; then
```

Comment:
It's clearer to set `PLATFORM=amd64|arm64` in a separate `case` block to avoid duplicating a long command.

But you should probably set this in `ci.yaml` and set the variables if they're unset so `.github/scripts/X` can still be run outside GHA, mostly for testing/dev purposes.